pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,390 @@
+	})
+	t.Run("restart deployment works successfully with whole operator(operator, non operator)", func(t *testing.T) {
+		//given
+		newClient, fakeClient := NewFakeClients(t, toolchainCluster, hostDeployment, hostPod, regServDeployment, noisePod)
+		ctx := clicontext.NewCommandContext(term, newClient)
 
-		// then
-		require.Error(t, err)
-		assert.Equal(t, 0, numberOfUpdateCalls)
-		AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 3)
-	}
-}
+		//when
+		err := restartDeployment(ctx, fakeClient, "toolchain-host-operator", nil, testIOStreams,
+			func(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+				require.Equal(t, testIOStreams, ioStreams)
+				require.Nil(t, f)
+				return nil
+			}, func(ctx *clicontext.CommandContext, deployment appsv1.Deployment, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error {
+				require.Equal(t, testIOStreams, ioStreams)
+				require.Nil(t, f)
+				return nil
+			})
+
+		//then
+		require.NoError(t, err)
+		//checking the flow for operator deployments
+		require.Contains(t, term.Output(), "Fetching the current Operator and non-Operator deployments of the operator in")
+		require.Contains(t, term.Output(), "Proceeding to delete the Pods of")
+		require.Contains(t, term.Output(), "Listing the pods to be deleted")
+		require.Contains(t, term.Output(), "Starting to delete the pods")
+		err = fakeClient.Get(ctx, test.NamespacedName("toolchain-host-operator", "host-operator-controller-manager"), actualPod)
+		//pods are actually deleted
+		require.True(t, apierror.IsNotFound(err))
+		require.Contains(t, term.Output(), "Checking the status of the deleted pod's deployment")
+		//checking the flow for non-operator deployments
+		require.Contains(t, term.Output(), "Proceeding to restart the non-operator deployment")
+		require.Contains(t, term.Output(), "Checking the status of the rolled out deployment")
+	})
 
-func TestRestartHostOperator(t *testing.T) {
-	// given
-	SetFileConfig(t, Host())
-	term := NewFakeTerminalWithResponse("") // it should not read the input
-	cfg, err := configuration.LoadClusterConfig(term, "host")
-	require.NoError(t, err)
-	namespacedName := types.NamespacedName{
-		Namespace: "toolchain-host-operator",
-		Name:      "host-operator-controller-manager",
-	}
+	t.Run("restart deployment works successfully when only operator based deployment", func(t *testing.T) {
+		//given
+		newClient, fakeClient := NewFakeClients(t, toolchainCluster, hostDeployment, hostPod)
+		ctx := clicontext.NewCommandContext(term, newClient)
+
+		//when
+		err := restartDeployment(ctx, fakeClient, "toolchain-host-operator", nil, testIOStreams,
+			func(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+				require.Equal(t, testIOStreams, ioStreams)
+				require.Nil(t, f)
```

Comment:
there is not much point in verifying these values, you should rather intercept the calls and check that it was really called with the expected values